dirsrvtests/tests/tickets/ticket2058_test.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import pytest
+import os
+import subprocess
+import time
+import ldap
+from lib389._constants import *
+from lib389.topologies import topology_i2 as topo
+
+DEBUGGING = os.getenv("DEBUGGING", default=False)
+if DEBUGGING:
+    logging.getLogger(__name__).setLevel(logging.DEBUG)
+else:
+    logging.getLogger(__name__).setLevel(logging.INFO)
+log = logging.getLogger(__name__)
+
+def run(cmd):
+  rc = subprocess.run(cmd.split(), capture_output=True)
+  if (rc.returncode != 0):
+      log.error(f"Command {cmd} failed: {rc}")
+      assert False
+  return rc
+
+def check_single_keep_alive(topo,instance,dn,expected):
+    # check that keep alive entry does not exists on master1
+    try:
+        ent = topo.ins[instance].getEntry(dn, ldap.SCOPE_BASE, "(objectclass=ldapsubentry)", ['nsUniqueId', 'modifierTimestamp'])
```

Comment:
Please, use direct python-ldap calls if you need something specific.
https://www.python-ldap.org/en/python-ldap-3.3.0/reference/ldap.html#ldap.LDAPObject.search_s

We plan to deprecate 'Entry' object some time soon. 

P.S. we also can add an interface for getting 'keep alive' entries to lib389 and use it from there.

Reply:
Well, if we have a keepalive we want to be able to check, we probably could add a lib389 dsldapobject type for it to make this a bit clearer...

Reply:
I get DeprecationWarning when using search_s  ...
DsLdapObject seems the way to go these days. that said creating a class for keepalive entries seems an overkill,
 as for now we only check their presence (and I am not sure we should really expose them in lib389 API)
My feeling is that we should rather provide a generic way to get a set of entries ...

Reply:
Yeah, it's been asked about before, to have a way to get a single entry by dn, or to do an objectClass=* style search. Maybe I should let go of my ideals and create it :)

Reply:
Getting a single object sound fine but beware to the filter:
   subentries are hidden if (objectclass=subentry) is not an explicit part of the filter
   so the filter should rather be:   (|(objectclass=*)(objectclass=subentry)) 

[progier@linux-3 slapd]$ ldapsearch -LLL -Y EXTERNAL -H ldapi://$S1 -b "cn=repl keep alive 1,dc=example,dc=com"  "(objectclass=*)" dn 2>/dev/null
[progier@linux-3 slapd]$ ldapsearch -LLL -Y EXTERNAL -H ldapi://$S1 -b "cn=repl keep alive 1,dc=example,dc=com"  "(|(objectclass=*)(objectclass=ldapsubentry))" dn 2>/dev/null
dn: cn=repl keep alive 1,dc=example,dc=com

[progier@linux-3 slapd]$

Reply:
Sorry, In the previous message I meant objectclass=ldapsubentry rather than objectclass=subentry !